_source/_posts/2024-01-24-spring-boot-authorization.md

```diff
@@ -0,0 +1,485 @@
+
+OpenID Connect provides authentication built on top of OAuth 2.0, and information about the authentication performed is returned in an ID Token with JWT format (JSON Web Token). In abstract, the protocol defines a client role or Relying Party that sends a request to the OpenID Provider, which in turn authenticates the end-user and obtains authorization, returning an ID Token and Access Token to the Relying Party.
+
+## Authorization in a Spring Boot API
+
+After the year 2020, in Buenos Aires, many bars and restaurants implemented a digital menu with a QR code that translates to a public document, for having an updated prices list. Prices change so often due to inflation, and updating a physical menu seems tedious and costly. Still, restaurant tables are managed with a software application, which includes a menu management module.
+
+For learning purposes, let's assume you have built a Spring Boot menu API that must be secured, so only authorized users can perform requests to its endpoints. Now you are going to implement authorization for the API with OAuth2 2.0 and Auth0. Start by doing a checkout of the API repository, which already implements basic request handling:
+
+```shell
+git clone https://github.com/indiepopart/spring-menu-api.git
+```
+
+The repository contains two project folders, `start` and `demo`. The bare bones menu API is a Gradle project in the `start` folder, open it with your favorite IDE. If you would rather skip the step-by-step security configuration and just run the final `demo` project, follow the instructions in the [README](https://github.com/indiepopart/spring-menu-api).
+
+Sign up at [Auth0](https://auth0.com/signup) and install the [Auth0 CLI](https://github.com/auth0/auth0-cli). Then in the command line run:
+
+```shell
+auth0 login
+```
+
+The command output will display a device confirmation code and open a browser session to activate the device.
+
+You don't need to create a client application for your API if not using opaque tokens. But you must register the API within your tenant, you can do it using Auth0 CLI:
+
+```shell
+auth0 apis create \
+  --name "Menu API" \
+  --identifier https://menu-api.okta.com \
+  --scopes "create:items,update:items,delete:items" \
+  --offline-access=false
+```
+
+The scopes `create:items`, `update:items`, `delete:items` will be required ahead in the tutorial. Add the `okta-spring-boot-starter` dependency:
+
+```groovy
+// build.gradle
+dependencies {
+    ...
+    implementation 'com.okta.spring:okta-spring-boot-starter:3.0.6'
+    ...
+}
+```
+As the `menu-api` must be configured as an OAuth2 resource server, add the following properties:
+
+```properties
+# application.properties
+okta.oauth2.issuer=https://<your-auth0-domain>/
+okta.oauth2.audience=https://menu-api.okta.com
+```
+
+```shell
+auth0 tenants list
```

Comment:
what is this command for? If its to get the issuer, please add some text with explanation

Reply:
Added text.